### ISSUE TITLE
Functions as input for the scale parameter in linear_extrude

### ISSUE THREAD

**Issue by Nexusnui** [openscad/openscad#5148]
**Is your feature request related to a problem? Please describe.**
Sometimes I want to extrude a 2D shape and want it to transiton in a curve.

**Describe the solution you'd like**
Let the scale parameter accept a function as input. The a provided function should accept either a progress or heigth parameter as input and return a scale for that heigth.

**Additional context**
I have implemented a version of this in the library "Variable Extrude" on Thingiverse, but a native version would be an improvement.
[The Library on Thingiverse](https://www.thingiverse.com/thing:6629632)

Some example pictures using `sin((z*360+45)*2)*0.25+0.75` as a scaling function, where z is the progress variable:
Square only scaled on x axis:
![image](https://github.com/openscad/openscad/assets/131207888/06c64d3d-5e10-4898-95ff-d233c2803482)
Square  scaled on both axes:
![image](https://github.com/openscad/openscad/assets/131207888/4d43e2d6-fff7-47f6-a052-aad3167be529)
Circle only scaled on x axis:
![image](https://github.com/openscad/openscad/assets/131207888/971f0280-025d-4862-aa49-b27e6d9ddd06)
Circle scaled on both axes:
![image](https://github.com/openscad/openscad/assets/131207888/21f3dd83-b0c2-43f3-9fa6-fbbf0e9725f8)

Some other examples that also use the twist parameter:
![image](https://github.com/openscad/openscad/assets/131207888/80e0630c-185b-47da-8a51-646249f88fc8)
![image](https://github.com/openscad/openscad/assets/131207888/aaf79139-3c37-47f5-8baf-3788ac7e1d1f)

**Comment by Nexusnui** [comment]
This might be interesting, since this could be used for other manipulations, but I am not sure how slices should be generated, if top and bottom are different shapes(wich is possible with your suggestion).
The function should probably take take two arguments total_heigth and current_height.
I have not updated the file posted on Thingiverse yet, but I have changed the function to accept two parameters.

**Comment by p-v-n** [comment]
Yeah, the above is just a rough suggestion to illustrate the idea.
Supplying total_height and current_height makes perfect sense, and would allow the function to produce the shape it is designed to.

**Comment by gsohler** [comment]
This is yet already possible with PythonSCAD (pythonscad.org)

https://imgur.com/a/MmxRMFk


[image: image.png]


Restriction here is that each slice should have same number of points
If you cannot satisfy this restriction, PythonSCAD has a loft function, too.

**Comment by jordanbrown0** [comment]
It's also not that hard to DIY a linear extrude that consumes a point list, and then you can do whatever you like to the point list on each layer.  If you're interested, send e-mail on the mailing list and I'll post a sample implementation.
